hash.go

```diff
@@ -38,6 +39,29 @@ func TreeHash(val interface{}) ([32]byte, error) {
 	return paddedOutput, nil
 }
 
+// SigningRoot returns the signed message for the value passed after truncating
```

Comment:
This comment is lacking. Why does it truncate? What length does it truncate to? 
What is the value that I'm supposed to use this with?